### PR TITLE
chore(docs): update multitrack-progress.md code examples

### PR DESCRIPTION
**Pull request by nagyszili**
update examples to a working version

### CODE REVIEW

File: docs/docs/guides/multitrack-progress.md [hunk 1]
```diff
@@ -96,8 +96,9 @@ const setProgress = useProgressStateStore.getState().setProgress;
 export const PlaybackService = async function() {
   TrackPlayer.addEventListener(Event.PlaybackProgressUpdated, async () => {
     // get the position and currently playing track.
-    const position = TrackPlayer.getPosition();
-    const track = TrackPlayer.getCurrentTrack();
+    const position = await TrackPlayer.getPosition();
+    const trackIndex = await TrackPlayer.getCurrentTrack();
+    const track = await TrackPlayer.getTrack(trackIndex || 0);
```
Comment (jspizziri) [inline docs/docs/guides/multitrack-progress.md:101]: These two lines can now be simplified to:

```ts
const track = await TrackPlayer.getActiveTrack();
```

**Review by jspizziri — CHANGES_REQUESTED** [review]
Thanks for flagging this. We should update it to reflect the new API's available in the `next` release.

Thanks!

File: docs/docs/guides/multitrack-progress.md [hunk 2]
```diff
@@ -96,8 +96,9 @@ const setProgress = useProgressStateStore.getState().setProgress;
 export const PlaybackService = async function() {
   TrackPlayer.addEventListener(Event.PlaybackProgressUpdated, async () => {
     // get the position and currently playing track.
-    const position = TrackPlayer.getPosition();
-    const track = TrackPlayer.getCurrentTrack();
+    const position = await TrackPlayer.getPosition();
```
[the remark this answers is not in the document]
Comment (puckey) [reply docs/docs/guides/multitrack-progress.md:99]: Isn’t the event listener callback called with an event object containing progress info?

File: docs/docs/guides/multitrack-progress.md (hunk 1 above)
Comment (nagyszili) [reply docs/docs/guides/multitrack-progress.md]: done

File: docs/docs/guides/multitrack-progress.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (nagyszili) [reply docs/docs/guides/multitrack-progress.md]: you'r right

File: docs/versioned_docs/version-3.2/guides/multitrack-progress.md [hunk 3]
```diff
@@ -94,12 +94,12 @@ import { useProgressStateStore } from '../store';
 const setProgress = useProgressStateStore.getState().setProgress;
 
 export const PlaybackService = async function() {
-  TrackPlayer.addEventListener(Event.PlaybackProgressUpdated, async () => {
+  TrackPlayer.addEventListener(Event.PlaybackProgressUpdated, async (event) => {
     // get the position and currently playing track.
-    const position = TrackPlayer.getPosition();
-    const track = TrackPlayer.getCurrentTrack();
+    const track = await TrackPlayer.getActiveTrack();
```
Comment (jspizziri) [inline docs/versioned_docs/version-3.2/guides/multitrack-progress.md:99]: `getProgress` isn't available in `v3.2` (which this doc is for) so we don't want to update this one. It should stay as is I believe. same with `getActiveTrack`.

File: docs/docs/guides/multitrack-progress.md [hunk 4]
```diff
@@ -94,12 +94,11 @@ import { useProgressStateStore } from '../store';
 const setProgress = useProgressStateStore.getState().setProgress;
 
 export const PlaybackService = async function() {
-  TrackPlayer.addEventListener(Event.PlaybackProgressUpdated, async () => {
+  TrackPlayer.addEventListener(Event.PlaybackProgressUpdated, async (event) => {
     // get the position and currently playing track.
-    const position = TrackPlayer.getPosition();
-    const track = TrackPlayer.getCurrentTrack();
+    const track = await TrackPlayer.getActiveTrack();
```
Comment (jspizziri) [inline docs/docs/guides/multitrack-progress.md:99]: We don't want to just get the active track here, as it's possible it's unloaded by this point. We should fetch it based on the ID in the event payload. I'd recommend something like my original comment.

**Review by jspizziri — CHANGES_REQUESTED** [review]

**Review by jspizziri — APPROVED** [review]

**Comment by jspizziri** [comment]
@nagyszili thanks for the help with the docs! Good stuff

**Pull request merged**